department-of-veterans-affairs/caseflow
# HunJerBAH/APPEALS-13375

Resolves [APPEALS-13375](https://vajira.max.gov/browse/APPEALS-13375)

### Description
Created cancel action for the Final Notification Letter Task

### Acceptance Criteria
- [ ] when the user selects the cancel action from the action dropdown, a modal appeals asking the user to cancel the task
- [ ] If cancel is confirmed, the “Send Final Notification Letter” task is cancelled
- [ ] If cancel is confirmed, Task moves from Currently Active Tasks section to Case Timeline and contains the following information
- [ ] Date of occurrence
- [ ] Assigned to: <user or organization>
- [ ] Assigned by: <user who completed action to assign task to COB>
- [ ] Cancelled by: <user who completed Send Final Notification Letter task>
- [ ] Task Instructions: <user input in the Proceed to final notification letter modal>


### Testing Plan
1. Go to the sign in page and sign in as a COB User.
![image](https://user-images.githubusercontent.com/99915461/222458768-6cd14c56-3102-48c7-a6e6-677989c8a97f.png)
2. Navigate to the Queue app.
3. Click on **switch views** followed by **navigate to COB organization**.
![image](https://user-images.githubusercontent.com/99915461/222459065-1c5544c5-a4ea-4f62-8be0-700843d33bcc.png)

4. Select a **Inital letter task** from the COB Organization table.
![image](https://user-images.githubusercontent.com/99915461/222459277-3f80fa32-9a6f-4067-aeed-152cb2e78647.png)

5. Click the **action** dropdown and select **proceed to final notification letter** from the dropdown.
![image](https://user-images.githubusercontent.com/99915461/222459469-f98eb9f9-1207-440a-b60b-542a25c1a49f.png)

6. Fill out the instructions and click **proceed to final letter**.
![image](https://user-images.githubusercontent.com/99915461/222459580-03771e4f-9251-4e4e-95d5-033e64cc166a.png)

7. You should end up on the Organization page and greeted with a green success banner. 
![image](https://user-images.githubusercontent.com/99915461/222459774-b6e7d7eb-c03a-4d9c-aa75-946de09d2088.png)

8. Find the **final notification letter task** in the table and click on it to navigate to the Case Details Page for that appeal.
9. Click the **action** dropdown and select **cancel task**.
![image](https://user-images.githubusercontent.com/99915461/222460106-c7f3331f-cdf0-4f8e-a4a2-f6cb70b9122a.png)

10. Fill out the form and select **submit**.
![image](https://user-images.githubusercontent.com/99915461/222460240-f172daa5-5cd4-4589-b73a-3a99b29a7be6.png)

11. You should be greeted by a green success banner. 
![image](https://user-images.githubusercontent.com/99915461/222460330-e824736c-ac27-46c8-bc43-b3dc2edf6666.png)

12. Click the back arrow to navigate back to the Case Details Page. There should be no action dropdown. If you scroll to the bottom of the page, you should see the Final task in the cancelled timeline with the instructions.
![image](https://user-images.githubusercontent.com/99915461/222460524-8e469510-c68a-4b0e-bc38-e7fb763a34bb.png)
![image](https://user-images.githubusercontent.com/99915461/222460704-27086ba1-7e31-4e17-ae5b-65b7da56f8f4.png)
![image](https://user-images.githubusercontent.com/99915461/222460731-c1478d75-0ba4-479a-9a57-7e2f35513ac6.png)
